src/SolutionAcceleration.f90

```diff
@@ -0,0 +1,395 @@
+!> @param solver Solver to take step with
+!> @param x_new New solution iterate and under-relaxed / accelerated return vector
+!>
+SUBROUTINE step_RelaxedPicardType(solver,x_new)
+  CLASS(RelaxedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x_new
+
+  REQUIRE(x_new%n == solver%n)
+
+  !Update iteration counter
+  solver%s=solver%s+1
+
+  IF(solver%s >= solver%start) THEN
+    !Get under-relaxed solution using mixing parameter
+    CALL BLAS_scal(x_new,solver%alpha)
+    CALL BLAS_axpy(solver%x(1),x_new,1.0_SRK-solver%alpha)
+  ENDIF
+  CALL BLAS_copy(x_new,solver%x(1))
+
+ENDSUBROUTINE step_RelaxedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Reset the initial iterate for the solver
+!> @param solver The solver to act on
+!> @param x Initial iterate solve is starting from
+!>
+SUBROUTINE reset_RelaxedPicardType(solver,x)
+  CLASS(RelaxedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x
+
+  REQUIRE(x%n == solver%n)
+
+  !Reset iteration counter
+  solver%s=0_SIK
+
+  CALL solver%setInitial(x)
+
+ENDSUBROUTINE reset_RelaxedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Clears the modified picard Type
+!> @param solver The solver to act on
+!>
+SUBROUTINE clear_ModifiedPicardType(solver)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+
+  CALL solver%clear_picardbase()
+
+  IF(ALLOCATED(solver%tmpvec)) THEN
+    CALL solver%tmpvec%clear()
+    DEALLOCATE(solver%tmpvec)
+  ENDIF
+  solver%isInit=.FALSE.
+ENDSUBROUTINE clear_ModifiedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Sets the initial vector for relaxation
+!> @param solver Solver to take step with
+!> @param x_new New solution iterate and under-relaxed / accelerated return vector
+!>
+!> This sets the initial vector into internal storage.  Also initializes solver%x
+!> and tmpvec based on the type that is passed in.
+!>
+SUBROUTINE setInitial_ModifiedPicardType(solver,x)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x
+
+  CALL solver%setInitial_picardBase(x)
+
+  IF(.NOT. ALLOCATED(solver%tmpvec)) CALL VectorResembleAlloc(solver%tmpvec,x)
+
+ENDSUBROUTINE setInitial_ModifiedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Performs a single step of Modified Picard acting on the input solution vector.
+!> @param solver Solver to take step with
+!> @param x_new New solution iterate and under-relaxed / accelerated return vector
+!>
+SUBROUTINE step_ModifiedPicardType(solver,x_new)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x_new
+
+  REQUIRE(x_new%n == solver%n)
+
+  !Update iteration counter
+  solver%s=solver%s+1
+
+  CALL BLAS_copy(x_new,solver%tmpvec)
```

Comment:
Is it required that `setInital` get called before `step`?  If not, then I think this call might fail due to `%tmpvec` not yet being allocated

Reply:
Yes, setInitial needs to be called first.  Honestly, this is consistent with the behavior of Anderson only you had to call reset first in Anderson.

Reply:
Ok... if we're going to require `setInitial` to be called, then `step` should really have some sort of DBC statement ensuring that the vector is set up.

That DBC statement should really be in the other 2 classes too, even if it's not strictly required.  We should ensure the same set of calls for all classes.

Reply:
Added